exp/api/v1beta1/azuremachinepool_default.go

```diff
@@ -72,18 +72,14 @@ func (amp *AzureMachinePool) SetIdentityDefaults(subscriptionID string) {
 		return
 	}
 	if amp.Spec.Identity == infrav1.VMIdentitySystemAssigned {
-		if amp.Spec.RoleAssignmentName == "" {
-			amp.Spec.RoleAssignmentName = string(uuid.NewUUID())
-		}
```

Comment:
Should we be preserving this workflow where the `amp.Spec.RoleAssignmentName == ""` if empty will be assigned a new `string(uuid.NewUUID())` ?

Reply:
We used to set `amp.Spec.RoleAssignmentName` to a new UUID if it's empty so we can set `amp.Spec.SystemAssignedIdentityRole.Name` to its value. We never intend to use `amp.Spec.RoleAssignmentName` as it is a deprecated field. In the new code, we are just checking if the user did indeed set `amp.Spec.RoleAssignmentName`. If not, we set `amp.Spec.SystemAssignedIdentityRole.Name` to a new UUID, which is the new field that replaces the deprecated one.